File: edukate-auth/src/main/java/io/github/sanyavertolet/edukate/auth/configs/RoleHierarchyConfiguration.java
package io.github.sanyavertolet.edukate.auth.configs;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.security.access.hierarchicalroles.RoleHierarchy;
import org.springframework.security.access.hierarchicalroles.RoleHierarchyImpl;

@Configuration
public class RoleHierarchyConfiguration {
    @Bean
    public RoleHierarchy roleHierarchy() {
        return RoleHierarchyImpl.fromHierarchy("ROLE_ADMIN > ROLE_MODERATOR > ROLE_USER");
    }
}

File: edukate-common/src/main/kotlin/io/github/sanyavertolet/edukate/common/configs/RoleHierarchyConfiguration.kt
package io.github.sanyavertolet.edukate.common.configs

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.security.access.hierarchicalroles.RoleHierarchy
import org.springframework.security.access.hierarchicalroles.RoleHierarchyImpl

@Configuration
class RoleHierarchyConfiguration {
     // todo: this does not seem to work at all
    @Bean fun roleHierarchy(): RoleHierarchy = RoleHierarchyImpl.withDefaultRolePrefix()
        .role("ADMIN").implies("MODERATOR")
        .role("MODERATOR").implies("USER")
        .build()
}
